app.js

```diff
@@ -0,0 +1,343 @@
+        compIcon = COMP_DEFAULT_ICON
+
+
+    }
+
+    return tossWinner
+
+}
+
+
+
+
+function createMatrix(size) {
+
+    const matrix = []
+
+    valueIndicesMatch = {}
+
+
+    let prevNum = 1
+
+    for (let i = 0; i < size; i++) {
+
+        const row = []
+
+        for (let j = 0; j < size; j++) {
+
+            row.push(prevNum)
+            remainingPlaces.push(prevNum) //updating the choices available for user/computer to opt from
+            valueIndicesMatch[prevNum] = [i, j]
+            prevNum += 1
+        }
+
+        matrix.push(row)
+
+    }
+
+    showMatrix(matrix)
+
+    return matrix
+}
+
+// createMatrix(4)
+
+
+function showMatrix(matrixToBePrinted) {
+
+    for (let row of matrixToBePrinted) {
+        console.log(row.join(' | '))
+    }
+    console.log('\n')
+}
+
+
+
+
+
+function userMarkingInput(remainingPlaces) {
+
+    let num = parseInt(prompt(`your chance please select only from : ${remainingPlaces}`))
+
+    while (!remainingPlaces.includes(num)) {
+        num = parseInt(prompt(`You chose  : ${num}. This input is not valid, please enter options available from  : ${remainingPlaces}`))
+    }
+    return num
+
+}
+
+
+
+
+// console.log(createMatrix(matrixSize))
+
+
+// const remainingPlaces = [1, 2, 3, 4, 4]
+// prompt(`your chance please select only from : ${remainingPlaces}`)
+
+
+function markChoiceInMatrix(inputNum, matrix, inputFrom) {
+
+
+    const icon = inputFrom === USER ? userIcon : compIcon
+
+    const index = remainingPlaces.indexOf(inputNum)
+
+    remainingPlaces.splice(index, 1)
+
+
+    const inputIndicesMatrix = valueIndicesMatch[inputNum]
+
+    matrix[inputIndicesMatrix[0]][inputIndicesMatrix[1]] = icon
+
+    // console.log(matrix)
+    // console.log(inputIndicesMatrix)
+    showMatrix(matrix)
+
+}
+
+function checkWinner(matrix, size) {
+
+    const firstDiagonal = new Set()
+    const secondDiagonal = new Set()
+
+    for (let i = 0; i < size; i++) {
+
+        const row = new Set()
+        const column = new Set()
+
+        firstDiagonal.add(matrix[i][i])
+        secondDiagonal.add(matrix[i][size - 1 - i])
+
+        for (let j = 0; j < size; j++) {
+
+            row.add(matrix[i][j])
+            column.add(matrix[j][i])
+
+
+
+        }
+
+
+        if (row.size === 1 | column.size === 1) {
+            return true
+        }
+
+
+
+    }
+    // console.log(firstDiagonal)
+    // console.log(secondDiagonal)
+
+    if (firstDiagonal.size === 1 | secondDiagonal.size === 1) {
+
+        return true
+    }
+
+    return false
+}
+
+
+
+
+
+const checkTie = (remainingelements) => remainingelements.length === 0 ? true : false
+
+
+
+
+function generateComputerInput(remainingelements) {
+
+    const index = Math.floor((Math.random() * remainingelements.length))
+
+    console.log(`Computer selected :  ${remainingelements[index]}`)
+
+    return remainingelements[index]
+}
+
+
+
+
+
+
+
+function gameFunctioning(tosswonby, matrix, matrixSizeUserInput, remainingPlaces) {
```

Comment:
- in this method, the entire `if` and `else` blocks are almost duplicated with only a small interchange of methods called
- can we remove this duplication?